src/ElmFormat.hs

```diff
@@ -109,16 +114,33 @@ data Destination
     | ToJson
 
 
-determineSource :: Bool -> Either [InputFileMessage] [FilePath] -> Either ErrorMessage Source
-determineSource stdin inputFiles =
+determineSource :: Bool -> Bool -> Maybe ElmVersion -> ElmVersion -> Either [InputFileMessage] [ElmFile] -> Either ErrorMessage Source
+determineSource stdin upgrade versionFlag defaultElmVersion inputFiles =
+    let
+        determineFile (FS.ElmFile fileDetectedVersion path) =
+            FS.ElmFile (upgradeVersion upgrade $ Maybe.fromMaybe fileDetectedVersion versionFlag) path
+    in
     case ( stdin, inputFiles ) of
         ( _, Left fileErrors ) -> Left $ BadInputFiles fileErrors
-        ( True, Right [] ) -> Right Stdin
+        ( True, Right [] ) -> Right $ Stdin $ upgradeVersion upgrade $ Maybe.fromMaybe defaultElmVersion versionFlag
         ( False, Right [] ) -> Left NoInputs
-        ( False, Right (first:rest) ) -> Right $ FromFiles first rest
+        ( False, Right (first:rest) ) -> Right $ FromFiles (determineFile first) (fmap determineFile rest)
         ( True, Right (_:_) ) -> Left TooManyInputs
 
 
+upgradeVersion :: Bool -> ElmVersion -> ElmVersion
+upgradeVersion upgrade version =
+    case (upgrade, version) of
+        (True, Elm_0_18) ->
+            Elm_0_18_Upgrade
+
+        (True, _) ->
```

Comment:
`checkUpgradeVersion` makes actually impossible to use something else than 0.19.

Still, the whole upgrade version code (check and upgrade) seems a little messy, but I'm not sure of the exact behavior we want, nor how to refactor this.